stats/stats.go

```diff
@@ -39,49 +50,79 @@ func InitStats() {
 	}
 
 	statsGlobal = &stats{
-		allMetrics: make(map[key]prom.Counter),
-		// This map will store prom.Counter instead of prom.Collector for now
-		// because typecasts are slow.
-		// A viable option would be to use a map of maps if we ever need
-		// to be able to store several metric types.
+		counters: make(map[key]prom.Counter),
+		gauges:   make(map[key]prom.Gauge),
 	}
 
 	statsGlobal.registerOwnMetrics()
 }
 
+func RegisterGauge(metricDesc *MetricDesc) {
+	maskPromGauge := prom.NewGauge(prom.GaugeOpts{
+		Namespace:   PromNamespace,
+		Subsystem:   metricDesc.Subsystem,
+		Name:        metricDesc.Name,
+		Help:        metricDesc.Help,
+		ConstLabels: map[string]string{"version": version.AppVersion},
+	})
+
+	keyInternal := key{PromNamespace, metricDesc.Subsystem, metricDesc.Name}
+	registerMetric(gaugeMetricType, keyInternal, maskPromGauge)
+}
+
 func RegisterCounter(metricDesc *MetricDesc) {
 	maskPromCounter := prom.NewCounter(prom.CounterOpts{
-		Namespace:   pipeline.PromNamespace,
+		Namespace:   PromNamespace,
 		Subsystem:   metricDesc.Subsystem,
 		Name:        metricDesc.Name,
 		Help:        metricDesc.Help,
 		ConstLabels: map[string]string{"version": version.AppVersion},
 	})
 
-	keyInternal := key{pipeline.PromNamespace, metricDesc.Subsystem, metricDesc.Name}
-	registerMetric(keyInternal, maskPromCounter)
+	keyInternal := key{PromNamespace, metricDesc.Subsystem, metricDesc.Name}
+	registerMetric(counterMetricType, keyInternal, maskPromCounter)
 }
 
-// Returns counter of given metric, than can be inc or dec.
+// Returns counter for a given metric.
 func GetCounter(subsystem, metricName string) prom.Counter {
 	checkStatsInitialized()
 
-	if val, ok := statsGlobal.allMetrics[getKey(subsystem, metricName)]; ok {
+	if val, ok := statsGlobal.counters[getKey(subsystem, metricName)]; ok {
 		return val
 	}
 
-	return statsGlobal.allMetrics[key{pipeline.PromNamespace, subsystem, metricName}]
+	return statsGlobal.counters[key{PromNamespace, subsystem, unknownCounter}]
 }
 
-func registerMetric(k key, metric prom.Counter) { // todo: support prom.Collector
+// Returns gauge for a given metric.
+func GetGauge(subsystem, metricName string) prom.Gauge {
```

Comment:
Comments should be like // GetGauge returns gauge for a given metric.

Reply:
done